app/src/src/screens/authentication/PasswordResetScreen.jsx

```diff
@@ -1,29 +1,144 @@
-import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import React, { useState } from 'react';
+import { AuthEndpoint } from '../../utils/endpoints';
+import { Image, StyleSheet, Text, View } from 'react-native';
+import { Button, TextInput } from 'react-native-paper';
+import { routes } from '../../utils/routes';
+import ErrorContext from '../../contexts/ErrorContext';
 import { StackNavigationProp } from '@react-navigation/stack';
 
+const initialState = {
+  /** @type {string | null} */
+  email: null,
+};
+
 /**
  * Renders the password reset screen.
  * @param {StackNavigationProp} navigation
  * @return {JSX.Element}
  * */
 const PasswordResetScreen = ({ navigation }) => {
+  const [email, setEmail] = useState(initialState.email);
+  const signInRedirect = () => {
+    // Navigate to the login screen
+    navigation.push(routes.login);
+  };
+
+  const emailSentRedirect = () => {
+    // Navigate to the password reset email sent screen
+    navigation.push(routes.emailSent);
+  };
+
+  const handlePasswordReset = async () => {
+    try {
+      await AuthEndpoint.passwordReset(email);
+      emailSentRedirect();
+    } catch (err) {
+      error.setMessage(err.message);
```

Comment:
so on ln6, I have this which is grayed out: import ErrorContext from '../../contexts/ErrorContext';
I feel like that line is supposed to define it... but it's grayed out... is there another way to define error?

Reply:
ln6 here only imports ErrorContext inside the file so it allows you to use ErrorContext, but it doesn't define it yet (it's grayed out because it's not used in the file yet).  An example of defining error is in the LoginScreen.jsx. There, it's defined like `const error = useContext(ErrorContext);` which also needs you to import `useContext` in ln1 like: `import React, { useContext, useState } from 'react';`

Reply:
Got it, thank you!